src/Microsoft.Extensions.Configuration.Ini/IniConfigurationExtensions.cs

```diff
@@ -78,36 +76,38 @@ public static IConfigurationBuilder AddIniFile(this IConfigurationBuilder builde
                 throw new ArgumentException(Resources.Error_InvalidFilePath, nameof(path));
             }
 
-            return AddIniFile(builder, source =>
-            {
-                source.Path = path;
-                source.Optional = optional;
-                source.ReloadOnChange = reloadOnChange;
-            });
+            return AddIniFile(builder, provider: null, path: path, optional: optional, reloadOnChange: reloadOnChange);
         }
 
         /// <summary>
         /// Adds a INI configuration source to <paramref name="builder"/>.
         /// </summary>
         /// <param name="builder">The <see cref="IConfigurationBuilder"/> to add to.</param>
-        /// <param name="configureSource">Configures the <see cref="IniConfigurationSource"/> to add.</param>
+        /// <param name="provider">The <see cref="IFileProvider"/> to use to access the file.</param>
+        /// <param name="path">Path relative to the base path stored in 
+        /// <see cref="IConfigurationBuilder.Properties"/> of <paramref name="builder"/>.</param>
+        /// <param name="optional">Whether the file is optional.</param>
+        /// <param name="reloadOnChange">Whether the configuration should be reloaded if the file changes.</param>
         /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
-        public static IConfigurationBuilder AddIniFile(
-            this IConfigurationBuilder builder,
-            Action<IniConfigurationSource> configureSource)
+        public static IConfigurationBuilder AddIniFile(this IConfigurationBuilder builder, IFileProvider provider, string path, bool optional, bool reloadOnChange)
         {
             if (builder == null)
```

Comment:
Nit: I don't think it would make any negative effect if we removed the null check for the builder from all the other overloads.